File: Server/ALTEA-Server/Models/Report.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ALTEA_Server.Models
{
    public class Report
    {
        public int Id { get; set; }
        public string? Topic { get; set; }
        public string? Description { get; set; }
        public string? configString { get; set; }
        public string? FileString { get; set; }
        public string? Path { get; set; }
        public ReportType? ReportType { get; set; }
        public int? DonorForeignKey { get; set; }
        [ForeignKey("DonorForeignKey")]
        public Donor? Donor { get; set; }
        public int? RecipientForeignKey { get; set; }
        [ForeignKey("RecipientForeignKey")]
        public Recipient? Recipient { get; set; }
    }
}

File: Client-App/src/app/models/report.ts
import { Donor } from "./user";
import { Recipient } from "./recipient";

export interface Report{

  id: number;
  description: string;
  topic: string;
  fileString: string;
  recipient: Recipient;
  recipientForeignKey: number;
  donor: Donor;
  donorForeignKey: number;
}
